### DISCUSSION TITLE
Feature: raise exception in case of typo in tab_panel name

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#2848] [Ideas / Feature Requests]
Consider:

```python
tab_list = ['Honey']
with ui.tabs() as tabs:
    [ui.tab(caption) for caption in tab_list]
with ui.row():
    with ui.tab_panels(tabs):
        with ui.tab_panel('Hunny'):  # should raise an exception because tabs contains 'Honey', not 'Hunny'
            pass
```

**Comment by rodja** [comment]
Sounds good. Would you like to create a pull request?

**Comment by falkoschindler** [comment]
Yes, one could expect an exception in this case. Technically, however, the `ui.tab`s _could_ be added later to the `ui.tabs` element - for whatever reason. So I'm not sure if raising an exception is _always_ the right thing to do.

**Comment by me21** [comment]
@falkoschindler could you share a sample code which shows what you mean?
I guess it could be a warning printed to the console instead of exception.

**Reply by falkoschindler** [reply]
```py
tabs = ui.tabs()
with ui.tab_panels(tabs):
    ui.tab_panel('Hunny')
with tabs:
    ui.tab('Honey')
```
😬
